lib/cfa/augeas_parser.rb

```diff
@@ -112,6 +122,11 @@ def delete(key)
       @data.reject! { |entry| entry[:key] == key }
     end
 
+    # @yieldparam block [Object] condition to satisfy
+    def delete_if(&_block)
```

Comment:
in fact I like more more flexible and consistent solution with matcher. I found it nicer and more reusable. So getting matcher instead of block.